docs/zkapps/tutorials/05-common-types-and-functions.mdx

```diff
@@ -0,0 +1,360 @@
+
+* [Circuit.if](../snarkyjs-reference/classes/Circuit#if)
+* [Circuit.switch](../snarkyjs-reference/classes/Circuit#switch)
+
+With these, you can write conditionals inside SnarkyJS.
+
+For example:
+
+```ts
+const input1 = Int64.from(10);
+const input2 = Int64.from(-15);
+
+const inputSum = input1.add(input2);
+
+const inputSumAbs = Circuit.if(inputSum.isPositive(), inputSum, inputSum.mul(Int64.minusOne));
+
+console.log(`inputSum: ${inputSum.toString()}`);
+console.log(`inputSumAbs: ${inputSumAbs.toString()}`);
+
+const input3 = Int64.from(22);
+
+const input1largest = input1.sub(input2).isPositive().and(input1.sub(input3).isPositive());
+const input2largest = input2.sub(input1).isPositive().and(input2.sub(input3).isPositive());
+const input3largest = input3.sub(input1).isPositive().and(input3.sub(input2).isPositive());
+
+const largest = Circuit.switch([ input1largest, input2largest, input3largest ], Int64, [ input1, input2, input3 ])
+
+console.log(`largest: ${largest.toString()}`);
+```
+
+With output:
+
+```sh
+inputSum: -5
+inputSumAbs: 5
+largest: 22
+```
+
+By inserting function calls, you can write circuits which usual if / else statements:
+
+```ts
+const result = Circuit.if(
+  <condition>, 
+  (() => { // TRUE
+    <code to use if true>
+    return <true-result>;
+  })(),
+  (() => { // FALSE
+    <code to use if false>
+    return <false-result>;
+  })()
+);
+```
+
+Note that unlike normal programming, because SnarkyJS under the hood is creating a zk cirucit, this is in fact executing both branches and selecting the one that matches the condition.
+
+## Assertions
+
+We've been using assertions throughout these tutorials already - but to clarify what they're doing, assertions are creating constraints in our circuit, that must hold true for a valid proof to be generated.
+
+Besides the `assertEquals` we've often used, there is also `assertTrue` available on the Bool class.
+
+## Merkle Trees
+
+Lastly, let's go over an example using [merkle trees](../snarkyjs-reference/modules/Experimental#merkletree). Merkle trees are very powerful, since they let us manage large amounts of data within a circuit. In the [project corresponding to this tutorial](https://github.com/es92/zkApp-examples/tree/main/05-common-types-and-functions/src), you can find a full reference for the example here. The contract can be found in [BasicMerkleTreeContract.ts](https://github.com/es92/zkApp-examples/blob/main/05-common-types-and-functions/src/BasicMerkleTreeContract.ts), and the example can be found in a section of [main.ts](https://github.com/es92/zkApp-examples/blob/main/05-common-types-and-functions/src/main.ts).
+
+Merkle trees are currently in the `Experimental` module of SnarkyJS. So start off by importing `Experimental`:
+```ts
+import {
+  ...
+  Experimental,
+  ...
+} from 'snarkyjs'
+```
+
+When using Merkle trees, you initialize an instance of the data structure on the "ui" side of your application at a particular height. 
```

Comment:
As a reader, this paragraph leaves me like "what i don't get it...and there's no example". Unless I really understand Merkle trees already, I'm not going to get it.

Maybe instead, this could start event higher level, or move this paragraph before the 2nd example, which I assume it's more related to.

This and the following paragraph feel like an opportunity to go more ELI5. @Trivo25 wrote the Merkle tree docs and might be able to help with a suggestion

Reply:
Updated - lmk what you think